internal/extproc/router/router_test.go

```diff
@@ -149,20 +149,84 @@ func TestRouter_selectBackendFromRule(t *testing.T) {
 
 	outSchema := filterapi.VersionedAPISchema{Name: filterapi.APISchemaOpenAI}
 
-	rule := &filterapi.RouteRule{
-		Backends: []filterapi.Backend{
-			{Name: "foo", Schema: outSchema, Weight: 1},
-			{Name: "bar", Schema: outSchema, Weight: 3},
+	tests := []struct {
+		name                           string
+		rule                           *filterapi.RouteRule
+		expectedSelectionsWithMaxDelta map[string]map[string]float64
```

Comment:
nit; could you use a struct rather than opaque map?

```suggestion
		expectedSelectionsWithMaxDelta map[string]struct {
			exp, maxDelta float64
		}
```

Reply:
good idea, updated